testing-framework/src/lib.rs

```diff
@@ -0,0 +1,337 @@
+// Copyright 2023 Polyphene.
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+mod error;
+
+use cid::{multihash::Code, Cid};
+use fil_builtin_actors_bundle::BUNDLE_CAR;
+use futures::executor::block_on;
+use fvm::{
+    engine::EnginePool,
+    executor::{ApplyKind, DefaultExecutor, Executor},
+    init_actor,
+    machine::{DefaultMachine, Manifest, NetworkConfig},
+    state_tree::{ActorState, StateTree},
+    system_actor,
+};
+use fvm_integration_tests::{
+    dummy::DummyExterns,
+    tester::{Account, IntegrationExecutor},
+};
+use fvm_ipld_blockstore::{Block, Blockstore, MemoryBlockstore};
+use fvm_ipld_car::load_car_unchecked;
+use fvm_ipld_encoding::{serde::Serialize, CborStore};
+use libsecp256k1::{PublicKey, SecretKey};
+use rand::SeedableRng;
+
+use error::{Error, WrapFVMError};
+use fvm_shared::{
+    address::Address, bigint::Zero, econ::TokenAmount, message::Message, state::StateTreeVersion,
+    version::NetworkVersion, ActorID, IPLD_RAW,
+};
+
+// TODO: document purpose.
+const EAM_ACTOR_ID: ActorID = 10;
+const NETWORK_VERSION: NetworkVersion = NetworkVersion::V18;
+const DEFAULT_BASE_FEE: u64 = 100;
+
+/// Main interface to test `Actor`s with Kythera.
+pub struct Tester {
+    // Builtin actors root Cid used in the Machine
+    builtin_actors: BuiltInActors,
+    // Custom code cid deployed by developer
+    code_cids: Vec<Cid>,
+    // State tree constructed before instantiating the Machine
+    state_tree: StateTree<MemoryBlockstore>,
+    // Account used for testing.
+    account: Account,
+    // Id of the main Actor deployed to be tested.
+    main_actor_id: Option<Vec<u8>>,
+}
+
+/// Built-in Actors that are deployed to the testing `StateTree`.
+struct BuiltInActors {
+    root: Cid,
+    manifest: Manifest,
+}
+
+/// WebAssembly Actor.
+pub struct WasmActor {
+    name: String,
+    code: Vec<u8>,
```

Comment:
Wouldn't some code be most likely be `&[u8]` instead ?

Reply:
have you started on this Thomas (after our conversation)? If not I can take it on.